apps/web/composables/useCategoryTemplate/useCategoryTemplate.ts

```diff
@@ -6,39 +6,80 @@ import type {
   SaveBlocks,
 } from '~/composables/useCategoryTemplate/types';
 import type { Block } from '@plentymarkets/shop-api';
+import type { FooterSettings } from '~/components/blocks/Footer/types';
 
 import homepageTemplateDataDe from './homepageTemplateDataDe.json';
 import homepageTemplateDataEn from './homepageTemplateDataEn.json';
 
 const useLocaleSpecificHomepageTemplate = (locale: string) =>
   locale === 'de' ? (homepageTemplateDataDe as Block[]) : (homepageTemplateDataEn as Block[]);
 
-export const useCategoryTemplate: UseCategoryTemplateReturn = () => {
-  const state = useState<UseCategoryTemplateState>('useCategoryTemplate', () => ({
+export const useCategoryTemplate: UseCategoryTemplateReturn = (blocks?: string) => {
+  const state = useState<UseCategoryTemplateState>(`useCategoryTemplate-${blocks}`, () => ({
     data: [],
     cleanData: [],
     categoryTemplateData: null,
     loading: false,
   }));
 
-  const getBlocks: GetBlocks = async (identifier, type) => {
+  const { $i18n } = useNuxtApp();
+
+  const getBlocksServer: GetBlocks = async (identifier, type, blocks?) => {
     state.value.loading = true;
-    const { data, error } = await useAsyncData(`${type}-${identifier}`, () =>
-      useSdk().plentysystems.getBlocks({ identifier, type }),
+
+    const { data, error } = await useAsyncData(`${type}-${identifier}-${blocks}`, () =>
+      useSdk().plentysystems.getBlocks({ identifier, type, ...(blocks ? { blocks } : {}) }),
     );
 
+    state.value.loading = false;
+
     if (error.value) {
       const { send } = useNotification();
-      send({ type: 'negative', message: error?.value?.message });
+      send({ type: 'negative', message: error.value.message });
+      return;
     }
 
-    state.value.loading = false;
+    let fetchedBlocks: Block[] = data?.value?.data ?? [];
 
-    if (!data?.value?.data.length && type === 'immutable') {
+    if (!fetchedBlocks.length && type === 'immutable') {
       const { $i18n } = useNuxtApp();
+      fetchedBlocks = useLocaleSpecificHomepageTemplate($i18n.locale.value);
+    }
+
+    const includeFooter = fetchedBlocks.find((block) => block.name === 'Footer');
```

Comment:
```suggestion
    const hasFooterBlock = fetchedBlocks.find((block) => block.name === 'Footer');
```
I think using a construction with has is a bit more idiomatic.